### DISCUSSION TITLE
Podman logs and priority level

### DISCUSSION THREAD

**Discussion by skoobasteeve** [containers/podman#18991] [Q&A] [answered]
Hello,
I'd like it if someone could help me understand podman's role in determining log priority levels when using `--log-driver=journald` combined with a syslog server. For some reason, 99% of the logs from my three containers (nextcloud-fpm, caddy, mariadb) are interpreted as "error" level by rsyslog. See attached examples as they appear on my LibreNMS server:

![Screenshot from 2023-06-24 13-32-30](https://github.com/containers/podman/assets/36998292/96d8ea7c-eb06-4f5b-8676-c99121c8c3f3)
![Screenshot from 2023-06-24 13-33-34](https://github.com/containers/podman/assets/36998292/27db5780-73ee-4ee5-949a-2bc19296cb59)

These appear to be standard info logs from the containers. How is rsyslog deciding that these logs should be error level? Not sure if I should be pursuing this with podman, rsyslog, or at the container level. Any guidance would be appreciated.

Thank you!

**Comment by Luap99** [comment, marked as answer]
The log driver only sees stdout and stderr from the container process.
stdout is logged at info level while stderr is logged at error level, see
https://github.com/containers/conmon/blob/2832e4f751939b4c63e34b2b6bcfbc103b1c7321/src/ctr_logging.c#L255-L258

**Reply by skoobasteeve** [reply]
So if I'm seeing these come in as `err` does that mean that the container is choosing so send this output to stderr instead of stdout?

**Reply by Luap99** [reply]
yes

**Reply by skoobasteeve** [reply]
Great, thanks for helping me understand. Is there a technical reason why a container author would choose to send seemingly routine messages to stderr instead of stdout?
